File: android/src/main/java/com/mycallkitwebrtc/plugin/app/SwiftFlutterCallkitIncomingPluginPlugin.java
package com.mycallkitwebrtc.plugin.app;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "SwiftFlutterCallkitIncomingPlugin")
public class SwiftFlutterCallkitIncomingPluginPlugin extends Plugin {

    private SwiftFlutterCallkitIncomingPlugin implementation = new SwiftFlutterCallkitIncomingPlugin();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }
}

File: ios/Plugin/SwiftFlutterCallkitIncomingPluginPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(SwiftFlutterCallkitIncomingPluginPlugin)
public class SwiftFlutterCallkitIncomingPluginPlugin: CAPPlugin {
    private let implementation = SwiftFlutterCallkitIncomingPlugin()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
